src/acl_command_queue.cpp

```diff
@@ -677,11 +677,13 @@ void acl_try_FastKernelRelaunch_ooo_queue_event_dependents(cl_event parent) {
 
     // Fast Kernel Relaunch: submitting is safe even though has dependency
     // Prior to submitting remove dependency
-    dependent->depend_on.erase(parent);
-    dependent_it = parent->depend_on_me.erase(dependent_it);
-    dependent_it--; // decrement it other wise we will skip an element
-    dependent->command_queue->num_commands_submitted++;
-    acl_submit_command(dependent);
+    int local_updates = acl_submit_command(dependent);
+    if (local_updates) {
+      dependent->depend_on.erase(parent);
+      dependent_it = parent->depend_on_me.erase(dependent_it);
+      dependent_it--; // decrement it otherwise we will skip an element
+      dependent->command_queue->num_commands_submitted++;
+    }
```

Comment:
Just curious, how does not returning num_update resolve the unit test hang?

Reply:
Talked to Sophie about this over a call. Fast Kernel Relaunch, is a feature that's okay to fail. In the previous version of the change, it would consider FKR failing as the new command failing to be submitted. However, it's okay if the new command fails to be submitted by FKR, since it'll be picked up again and submitted when its parent command (the one it depends on, which must exist since FKR happens in this case only) finishes.